rspirv/sr/autogen_instruction_structs.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// AUTOMATICALLY GENERATED from the SPIR-V JSON grammar:
+//   external/spirv.core.grammar.json.
+// DO NOT MODIFY!
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Extension {
+    name: String,
+}
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ExtInstImport {
+    name: String,
+}
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct MemoryModel {
+    addressing_model: spirv::AddressingModel,
+    memory_model: spirv::MemoryModel,
+}
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct EntryPoint {
+    execution_model: spirv::ExecutionModel,
+    entry_point: spirv::Word,
+    name: String,
+    interface: Vec<spirv::Word>,
+}
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ExecutionMode {
+    entry_point: spirv::Word,
+    mode: spirv::ExecutionMode,
+}
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Capability {
+    capability: spirv::Capability,
+}
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Function {
+    function_control: spirv::FunctionControl,
+    function_type: spirv::Word,
+}
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct FunctionParameter {}
```

Comment:
Do we still want to have separate ops for these function components? I think they should be folded into the function SR itself. I mean, `FunctionEnd` is useful for marking the end of a function in SPIR-V binary, but not useful for SR. Similarly for `Label`, it should be implicit for BasicBlock SR. Letting each pass processing Function or BasicBlock to worry about `FunctionEnd` and `Label` is not helpful.

Reply:
I agree, we'll need to set up a list of ops that don't need SR autogen generation, and which we are going to be converting to and from DR manually. We'll do it as a follow-up.